### ISSUE TITLE
Parameter name codemining: Selection on double-click off by one character

### ISSUE THREAD

**Issue by derBluemel** [angelozerr/jdt-codemining#10]
When I double-click on a parameter name or the parameter itself, the selection is set incorrectly.
It seems that the parameter name codemining is overlapping with the parameter itself by one character:

![bug](https://user-images.githubusercontent.com/1775306/37990014-50dab74c-3205-11e8-9353-6cff5f9361ee.gif)

I also changed the background color with an org.eclipse.swt.custom.LineBackgroundListener for the attached screenshot.

**Comment by angelozerr** [comment]
@derBluemel yes indeed today we have managed like this. Inlined content annotation takes place with GlyphMetrics width.  But the use of GlyphMetrics replace the character so we must redraw it.

To fix this issue, we must improve OS TextLayout. See my current experimentation with windows https://bugs.eclipse.org/bugs/show_bug.cgi?id=531769

**Comment by derBluemel** [comment]
@angelozerr 
OK! Thanks for your feedback. Doesn't seem to be an easy fix then... :(

On a side note: Why is the inlined annotation not affected by the LineBackgroundListener?
